test/controllers/movements/SummaryControllerSpec.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.movements
+
+import controllers.ControllerLayerSpec
+import forms.ConsignmentReferences
+import models.cache._
+import org.mockito.ArgumentMatchers.{any, anyString}
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito.{reset, verify, when}
+import play.api.libs.json.JsString
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import repository.MockCache
+import services.SubmissionService
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+import views.html.movement_confirmation_page
+import views.html.summary.{arrival_summary_page, departure_summary_page}
+
+import scala.concurrent.ExecutionContext.global
+import scala.concurrent.Future
+
+class SummaryControllerSpec extends ControllerLayerSpec with MockCache {
+
+  private val mockArrivalSummaryPage = mock[arrival_summary_page]
+  private val mockDepartureSummaryPage = mock[departure_summary_page]
+  private val mockMovementConfirmationPage = mock[movement_confirmation_page]
+
+  val submissionService: SubmissionService = mock[SubmissionService]
+
+  private def controller(answers: Answers = ArrivalAnswers()) =
+    new SummaryController(
+      SuccessfulAuth(),
+      ValidJourney(answers),
+      cache,
+      submissionService,
+      stubMessagesControllerComponents(),
+      mockArrivalSummaryPage,
+      mockDepartureSummaryPage,
+      mockMovementConfirmationPage
+    )(global)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+
+    when(mockArrivalSummaryPage.apply(any())(any(), any())).thenReturn(HtmlFormat.empty)
+    when(mockDepartureSummaryPage.apply(any())(any(), any())).thenReturn(HtmlFormat.empty)
+    when(mockMovementConfirmationPage.apply(any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(mockArrivalSummaryPage, mockDepartureSummaryPage, mockMovementConfirmationPage)
+
+    super.afterEach()
+  }
+
+  "Summary Controller" should {
+
+    "return 200 (OK)" when {
+
+      "GET displayPage is invoked without data in cache" in {
+
+        givenTheCacheIsEmpty()
+
+        val result = controller().displayPage()(getRequest)
+
+        status(result) mustBe OK
```

Comment:
Can we assert on what view was sent in the response please :)

Reply:
Done